### ISSUE TITLE
Improve InvertII

### ISSUE THREAD

**Issue by stelfrich** [imagej/imagej-ops#507]
[`InvertII`](https://github.com/imagej/imagej-ops/blob/dbbc347130ba9ee1b6659d9f0b5b20310960eb7f/src/main/java/net/imagej/ops/image/invert/InvertII.java) currently does inversion of `IterableInterval<I>`s solely based on the type of `I`.

As shortly discussed at the Fiji Learnathon, we might want to think about changing that behavior s.t. inversion can be based on the actual range of values in an `IterableInterval`:

```java
// Usage:
RandomAccessibleInterval<DoubleType> input;
image().invert(input, new DoubleType(10d), new DoubleType(100d));
  // Inverts values such that 10 maps to 100 and 100 maps to 10
image().invert(input);
  // Inverts based on the actual min and max of input
image().invert(input, new DoubleType().getMinValue(), new DoubleType().getMaxValue());
  // Inverts based only on the type's min and max
```

Any input on that topic? @imagejan 

See also https://github.com/knime-ip/knip/issues/470.

**Comment by tibuch** [comment]
```
image().inver(input);
```
Can get really slow if the whole image has to be scanned. Why not make inverting based on the type's min/max default? 

Then the user has to determine min/max by himself and it would be transparent why it takes longer.

**Comment by stelfrich** [comment]
From https://de.mathworks.com/help/images/ref/imcomplement.html?s_tid=gn_loc_drop:
> In the complement of an intensity [...] image, each pixel value is subtracted from the maximum pixel value supported by the class (or 1.0 for double-precision images) and the difference is used as the pixel value in the output image.

**Comment by stelfrich** [comment]
Development happening on https://github.com/imagej/imagej-ops/tree/improve-invert-ii

**Comment by stelfrich** [comment]
> Why not make inverting based on the type's min/max default?

Now that I have thought about it a bit, @tibuch, I'd favor your option. If we opt for the type-based default behavior, explicitly using min/max will look like this:

```java
ops().image().invert(input, ops().stats().min(input), ops().stats().max(input));
```

which seems more Ops-like than

```java
ops().image().invert(input, Util.getTypeFromInterval(input).getMinValue(), Util.getTypeFromInterval(input).getMaxValue());
```

**Comment by ctrueden** [comment]
@stelfrich If you are OK with it, I can have @gselzer finish up this work.

**Comment by stelfrich** [comment]
@ctrueden Sure! If you need an opinion, I'd be happy to provide one 😄

**Comment by ctrueden** [comment]
See PR #517 for current status on this.

**Comment by gselzer** [comment]
@ctrueden should this PR be closed now that #517 has been merged?
